### PR TITLE
WIP: Add a test for qgt

### PR DESCRIPTION
**Pull request by gcarleo**
This adds a simple test for the quantum geometric tensor (it should use the other testing facility for variational things, but it currently is...) 
The tests pass for real parameters, but for complex parameters they do not (at the moment...). 

To be investigated further!

### CODE REVIEW

File: test/variational/test_qgt.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+def GradPsi(pars,x):
+    return jax.vmap(GradPsiSingle,(None,0))(pars,x)
+
+
+
+PsiVec=Psi(vs.variables,all_states)
+GVec=GradPsi(vs.variables,all_states)
+
+#The exact qgt is <GradPsi|GradPsi> -<GradPsi|Psi><Psi|GradPsi>
+qgt_ex=(GVec.conjugate().transpose()@GVec) -(GVec.conjugate().transpose()@PsiVec)*(PsiVec.conjugate().transpose()@GVec)
```
[the remark this answers is not in the document]
Comment (attila-i-szabo) [reply test/variational/test_qgt.py:39]: I don't think that matters, the centring term should vanish if the wave function is normalised. I also think it should for a holomorphic C→C function too (e.g. 0 = d/dz* <Ψ|Ψ> = <dΨ/dz*|Ψ> because the second term has no z* derivative), so it would be good to check whether it actually does. 

Also, both `(GVec.conjugate().transpose()@PsiVec)` and `(PsiVec.conjugate().transpose()@GVec)` are just 1D vectors, so multiplying them like this gives a vector that broadcasts on the matrix, rather than the outer product matrix. However, this shouldn't matter as they're supposed to vanish, but there's definitely something fishy going on with this line

[the remark this answers is not in the document]
Comment (gcarleo) [reply test/variational/test_qgt.py:39]: erm yes, you are right, there should be an outer product....

[the remark this answers is not in the document]
Comment (PhilipVinc) [reply test/variational/test_qgt.py:39]: I just checked and `(GVec.conjugate().transpose()@PsiVec)` is numerically zero only for `dtype=float`.
It is not for `dtype=complex`.

[the remark this answers is not in the document]
Comment (gcarleo) [reply test/variational/test_qgt.py]: I think that also for R->C it is non zero, now I fixed the outer product and R->R and R->C both seem to work correctly, I still have issues with C->C

[the remark this answers is not in the document]
Comment (PhilipVinc) [reply test/variational/test_qgt.py]: yeah i can reproduce...

this is very curious.

**Comment by PhilipVinc** [comment]
I tried running this gist https://gist.github.com/PhilipVinc/6a22a5fc272fdc27b139429844657a60 
and it seems that there is a constant of 4 between the output of the `exact` qgt and our implementations.

File: test/variational/test_qgt.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (attila-i-szabo) [reply test/variational/test_qgt.py]: `Psi(pars,x)` is **not** a holomorphic function of `pars` because we divide through with `NormPsi` which is clearly not holomorphic, so `grad` will produce garbage. Taking the gradients of the unnormalised wave function and normalising afterwards fixes the error: https://gist.github.com/attila-i-szabo/fe35c1e874d9b6902d60812a6cea9db5

[the remark this answers is not in the document]
Comment (PhilipVinc) [reply test/variational/test_qgt.py]: I love you

[the remark this answers is not in the document]
Comment (gcarleo) [reply test/variational/test_qgt.py]: Great solve!
